### PR TITLE
Supporting custom labels/annotations on deployment/pod

### PR DESCRIPTION
**Pull request by jkroepke**
Signed-off-by: Jan-Otto Kröpke <joe@cloudeteer.de>

#### Overview

Added the ability to define custom annotations on deployment/pods

Tools like https://github.com/stakater/Reloader depends on that.

#### Checklist

- [x] Change log updated in `Chart.yaml` (see the contributing guide for details)
- [x] Chart version bumped in `Chart.yaml` (see the contributing guide for details)
- [x] Documentation regenerated by running `make docs`

### CODE REVIEW

**Comment by sagikazarmark** [comment]
@jkroepke this needs rebase as well

**Comment by jkroepke** [comment]
@sagikazarmark rebased.

**Comment by jkroepke** [comment]
@sagikazarmark can you please take a look here again?

**Comment by jkroepke** [comment]
@sagikazarmark A friendly reminder.

**Comment by jkroepke** [comment]
@sagikazarmark I would appreciate a review here, please let me know, if I can assist here.

File: charts/dex/values.yaml [hunk 1]
```diff
@@ -107,8 +107,15 @@ rbac:
   # If disabled, the operator is responsible for creating the necessary resources (ClusterRole and RoleBinding or CRD's)
   createClusterScoped: true
 
+# -- Annotations to be added to deployment.
+annotations: {}
```
Comment (sagikazarmark) [inline charts/dex/values.yaml:111]: How about `deploymentAnnotations` and `deploymentLabels`?

File: charts/dex/values.yaml [hunk 2]
```diff
@@ -214,8 +221,44 @@ serviceMonitor:
...
+  # -- Relabel configs to apply
+  # to samples before ingestion.
+  ## [Relabeling](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config)
+  relabelings: []
+  # - sourceLabels: [__meta_kubernetes_pod_node_name]
+  #   separator: ;
+  #   regex: ^(.*)$
+  #   targetLabel: nodename
+  #   replacement: $1
+  #   action: replace
```
Comment (sagikazarmark) [inline charts/dex/values.yaml:260]: Not a huge fan of mixing things in a PR, but I'm gonna let it pass. But please add it to the changelog. Thanks!

**Review by sagikazarmark — CHANGES_REQUESTED** [review]

File: charts/dex/values.yaml (hunk 2 above)
Comment (jkroepke) [reply charts/dex/values.yaml:260]: Seems like I do a merge mistake here on my local machine. If you want I can split the serviceMonitor logic into a different PR

Comment (jkroepke) [reply charts/dex/values.yaml:260]: Done

**Comment by jkroepke** [comment]
@sagikazarmark I address all your points.

File: charts/dex/ci/label-annotations-values.yaml [hunk 3]
```diff
@@ -6,5 +6,14 @@ config:
 
   enablePasswordDB: true
 
+annotations:
```
Comment (sagikazarmark) [inline charts/dex/ci/label-annotations-values.yaml:9]:
```suggestion
deploymentAnnotations:
```

File: charts/dex/ci/label-annotations-values.yaml [hunk 4]
```diff
@@ -6,5 +6,14 @@ config:
...
+  reloader.stakater.com/auto: "true"
+
 podAnnotations:
   vault.security.banzaicloud.io/vault-addr: "https://vault.vault:8200"
+
+labels:
```
Comment (sagikazarmark) [inline charts/dex/ci/label-annotations-values.yaml:15]:
```suggestion
deploymentLabels:
```

File: charts/dex/values.yaml [hunk 5]
```diff
@@ -107,8 +107,15 @@ rbac:
   # If disabled, the operator is responsible for creating the necessary resources (ClusterRole and RoleBinding or CRD's)
   createClusterScoped: true
 
+# -- Annotations to be added to deployment.
+deploymentAnnotations: {}
```
Comment (sagikazarmark) [inline charts/dex/values.yaml:111]:
```suggestion
deploymentAnnotations: {}

```

File: charts/dex/values.yaml [hunk 6]
```diff
@@ -107,8 +107,15 @@ rbac:
...
+# -- Labels to be added to deployment.
+deploymentLabels: {}
+
 # -- Annotations to be added to pods.
 podAnnotations: {}
+# -- Labels to be added to pods.
```
Comment (sagikazarmark) [inline charts/dex/values.yaml:117]:
```suggestion

# -- Labels to be added to pods.
```

**Review by sagikazarmark — CHANGES_REQUESTED** [review]
A few more nits. Other than those, it's good to go.

**Comment by jkroepke** [comment]
@sagikazarmark  Applied, thanks for review!

**Review by sagikazarmark — APPROVED** [review]

**Comment by sagikazarmark** [comment]
Thanks @jkroepke !

**Pull request merged**